app.test.js

```diff
@@ -337,22 +337,218 @@ describe("/api/articles", () => {
+                    expect(articles).toHaveLength(11)
+                    expect(articles).toBeSortedBy("votes", { descending: true })
+                    articles.forEach((article) => {
+                        expect(article).toEqual(
+                            expect.objectContaining({
+                                topic: "mitch",
+                                article_id: expect.any(Number),
+                                title: expect.any(String),
+                                author: expect.any(String),
+                                body: expect.any(String),
+                                created_at: expect.any(String),
+                                votes: expect.any(Number),
+                                comment_count: expect.any(Number)
+                            })
+                        )
+                    })
+                })
+            })
+            test("status: 200 responds with empty array if filter query does not yield any results", () => {
+                return request(app)
+                .get("/api/articles?topic=dogs")
+                .expect(200)
+                .then(({body : { articles }}) => {
+                    expect(articles).toBeInstanceOf(Array)
+                    expect(articles).toHaveLength(0)
+                })
+            })
+        })
+        describe("Error Handling", () => {
+            test("status: 400 responds with appropriate message if user attempts to sort_by invalid property", () => {
+                return request(app)
+                .get("/api/articles?sort_by=word_count")
+                .expect(400)
+                .then(({body}) => {
+                    expect(body.msg).toBe("Invalid query parameter")
+                })
+            })
+            test("status: 400 responds with appropriate message if user attempts to order using invalid syntax", () => {
+                return request(app)
+                .get("/api/articles?order=ascending")
+                .expect(400)
+                .then(({body}) => {
+                    expect(body.msg).toBe("Invalid query parameter")
+                })
+            })
+            test("status: 400 responds with appropriate message if query property does not exist (single query)", () => {
+                return request(app)
+                .get("/api/articles?sortby=votes")
+                .expect(400)
+                .then(({body}) => {
+                    expect(body.msg).toBe("Invalid query property")
+                })
+            })
+            test("status: 400 responds with appropriate message if query property does not exist (multiple queries)", () => {
+                return request(app)
+                .get("/api/articles?sortby=votes&order=asc")
+                .expect(400)
+                .then(({body}) => {
+                    expect(body.msg).toBe("Invalid query property")
+                })
+            })
```

Comment:
can we just add a test for :  Status 404. non-existent topic query, e.g. ?topic=bananas
please